### PR TITLE
[Merged by Bors] - libp2p upgrade + gossipsub interval fix

### PR DESCRIPTION
**Pull request by divagant-martian**
## Issue Addressed
Lighthouse gossiping late messages

## Proposed Changes
Point LH to our fork using tokio interval, which 1) works as expected 2) is more performant than the previous version that actually worked as expected
Upgrade libp2p 

## Additional Info
https://github.com/libp2p/rust-libp2p/issues/2497

### CODE REVIEW

File: beacon_node/lighthouse_network/Cargo.toml [hunk 1]
```diff
@@ -38,10 +38,11 @@ directory = { path = "../../common/directory" }
 regex = "1.3.9"
 strum = { version = "0.21.0", features = ["derive"] }
 superstruct = "0.4.0"
-open-metrics-client = "0.14.0"
+prometheus-client = "0.15.0"
 
 [dependencies.libp2p]
-version = "0.42.1"
+git = "https://github.com/sigp/rust-libp2p"
+branch = "libp2p-gossipsub-interval-hotfix"
```
Comment (AgeManning) [inline beacon_node/lighthouse_network/Cargo.toml]: We should tag the specific commit. 

I've been in trouble before where we've added to the branch or someone reverts back to here but that branch is different or cargo update got a different commit. I've learnt to use specific commits so we can't get in trouble here.

Comment (divagant-martian) [reply beacon_node/lighthouse_network/Cargo.toml]: done!

**Review by AgeManning — APPROVED** [review]
Nice. Lets get this guy in to save us from all the pain!

**Pull request closed without merging**